src/tangents.jl

```diff
@@ -813,7 +813,8 @@ same tangent twice and producing incorrect results.
 """
 require_tangent_cache(::Type{P}) where {P} = Val{!isbitstype(P)}()
```

Comment:
Maybe rename this to `require_SetToZeroCache` for consistency? 
```suggestion
require_SetToZeroCache(::Type{P}) where {P} = Val{!isbitstype(P)}()
```

Reply:
I don't think this is needed -- `require_tangent_cache` is a more general piece of functionality which applies to a range of functions (`increment!!`, `_dot`, etc). We're not currently making use of it in this PR, but we ought to do so in a follow-up PR.